components/search/search.service.ts

```diff
@@ -1,15 +1,17 @@
 import {readValue} from '../util/util';
 
 // Define useful types to avoid any.
-export type LookupFn<T> = (query:string) => Promise<T[]>
+export type LookupFn<T, U> = (query:any, initial?:boolean) => Promise<T[]>
```

Comment:
The value from `ngModel` is available in `query` parameter. But yes, maybe it would be better if we're consistently putting query (even though it's always empty in the initialization stage) into `query` parameter and use `inital` parameter for the initial `ngModel` value.